content/content.go

```diff
@@ -140,15 +141,38 @@ var rulesWithContentStorage = RulesWithContentStorage{
 	rules:            map[types.RuleID]*types.RuleContent{},
 }
 
+// RuleContentDirectoryTimeoutError is used, when the content directory is empty for too long time
+type RuleContentDirectoryTimeoutError struct{}
+
+func (e *RuleContentDirectoryTimeoutError) Error() string {
+	return "Content directory cache has been empty for too long time; timeout triggered"
+}
+
 // WaitForContentDirectoryToBeReady ensures the rule content directory is safe to read/write
-func WaitForContentDirectoryToBeReady() {
+func WaitForContentDirectoryToBeReady() error {
 	// according to the example in the official dock,
 	// lock is required here
 	if ruleContentDirectory == nil {
 		ruleContentDirectoryReady.L.Lock()
-		ruleContentDirectoryReady.Wait()
+
+		done := make(chan struct{})
+		go func() {
+			ruleContentDirectoryReady.Wait()
```

Comment:
Hi! I tried to add a print at the end of this function, `log.Info().Msg("Unlocking the lock")` at line 171, but it never get printed and the endpoints don't seem to work , what is it waiting for here?

Edit: approving, probably i just don't understand how to test it